src/transformers/models/fuyu/processing_fuyu.py

```diff
@@ -591,6 +575,111 @@ def __call__(
+            return (None, None)
+
+        def tokens_to_boxes(tokens, original_size):
+            while (pair := find_delimiters_pair(tokens, TOKEN_BBOX_OPEN_STRING, TOKEN_BBOX_CLOSE_STRING)) != (None, None):
+                start, end = pair
+                if end != start + 5:
+                    continue
+
+                # Retrieve transformed coordinates from tokens
+                coords = self.tokenizer.convert_ids_to_tokens(tokens[start+1:end])
+
+                # Scale back to original image size and multiply by 2
+                scale = scale_factor_to_fit(original_size)
+                top, left, bottom, right = [2 * int(float(c)/scale) for c in coords]
+
+                # Replace the IDs so they get detokenized right
+                replacement = f" <box>{top}, {left}, {bottom}, {right}</box>"
+                replacement = self.tokenizer.tokenize(replacement)[1:]
+                replacement = self.tokenizer.convert_tokens_to_ids(replacement)
+                replacement = torch.tensor(replacement).to(tokens)
+
+                tokens = torch.cat([tokens[:start], replacement, tokens[end+1:]], 0)
+            return tokens
+
+        def tokens_to_points(tokens, original_size):
+            while (pair := find_delimiters_pair(tokens, TOKEN_POINT_OPEN_STRING, TOKEN_POINT_CLOSE_STRING)) != (None, None):
+                start, end = pair
+                if end != start + 3:
+                    continue
+
+                # Retrieve transformed coordinates from tokens
+                coords = self.tokenizer.convert_ids_to_tokens(tokens[start+1:end])
+
+                # Scale back to original image size and multiply by 2
+                scale = scale_factor_to_fit(original_size)
+                x, y = [2 * int(float(c)/scale) for c in coords]
+
+                # Replace the IDs so they get detokenized right
+                replacement = f" <point>{x}, {y}</point>"
+                replacement = self.tokenizer.tokenize(replacement)[1:]
+                replacement = self.tokenizer.convert_tokens_to_ids(replacement)
+                replacement = torch.tensor(replacement).to(tokens)
+
+                tokens = torch.cat([tokens[:start], replacement, tokens[end+1:]], 0)
+            return tokens
+
+        if target_sizes is None:
+            target_sizes = ((self.image_processor.size["height"], self.image_processor.size["width"]),) * len(outputs)
+
+        if len(outputs) != len(target_sizes):
+            raise ValueError("Make sure that you pass in as many target sizes as output sequences")
+        if target_sizes.shape[1] != 2:
+            raise ValueError("Each element of target_sizes must contain the size (h, w) of each image of the batch")
+
+        results = None
+        for seq, size in zip(outputs, target_sizes):
+            seq = tokens_to_boxes(seq, size)
+            seq = tokens_to_points(seq, size)
+            seq = seq[None, :]
+            # TODO: what if sequence lengths vary?
```

Comment:
I haven't tested this yet, but if we use batch inference and there are different number of encoded boxes or points in the sequences, the tensor shapes will be different after these transformations. How can we deal with this case?

Reply:
Not sure what's best. I can see two ways: either we have a a way of padding the sequences to the max length then concatenating, or we return a list of variable length tensors. 

If I've understood the method correctly, this is postprocessing the model outputs and not producing the raw outputs which would be fed back to the model when generating the next token. In this case, do we need to return a batched tensor?

Reply:
I think returning a list of variable lengths tensors is fine, there's few things after generation that'd leverage a padded version of it. Just something like 
```
results = []
for seq, size in zip(outputs, target_sizes):
    seq = tokens_to_boxes(seq, size)
    seq = tokens_to_points(seq, size)
    results.append(seq)
```

Reply:
Yes, you are both right, thanks! I've checked in other post-processors that there's no need to keep the same input type if it's not necessary.